realm/realm-library/src/objectServer/java/io/realm/mongodb/Profile.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb;
+
+import javax.annotation.Nullable;
+
+import io.realm.internal.objectstore.OsSyncUser;
+
+public class Profile {
+    private final OsSyncUser osUser;
+
+    public Profile(OsSyncUser osUser) {
```

Comment:
This constructor shouldn't be public if possible. Also, we should probably send in the `User` class as a parameter as we can then expose it through a `getUser()` (for completeness, so we can always find the user/app from any class in the class hierachy)